### PR TITLE
Fix https://github.com/drighelli/SpatialExperiment/issues/135 -- thou…

### PR DESCRIPTION
**Pull request by lcolladotor**
…gh no new unit tests were added and might be useful

### CODE REVIEW

**Comment by lmweber** [comment]
Thanks a lot for finding this @lcolladotor , and for proposing the fix.

I have added a unit test and NEWS item plus version bump. @HelenaLC @drighelli could you please check this (described in more detail in #135 ), including the unit test I added above?

This is an urgent bugfix, since the current version of `read10xVisium()` is creating SPE objects with spatial coordinates in incorrect order in datasets with multiple samples. I think @lcolladotor 's solution above solves it, and the unit test should avoid new problems if we need to update / extend `read10xVisium()` again in the future.

I'll also create a separate PR for release branch so we can patch it into both release and devel.

**Comment by lcolladotor** [comment]
Awesome work Lukas! Thanks for improving the PR. I like your unit test too =) It's very clever as it avoid having to re-implement internals.

File: R/read10xVisium.R [hunk 1]
```diff
@@ -125,7 +125,7 @@ read10xVisium <- function(samples="",
     
     dir <- file.path(samples, "spatial")
     xyz <- file.path(
-        rep(dir, each = length(sids)), 
+        rep(dir, each = 2),
```
Comment (HelenaLC) [inline R/read10xVisium.R:128]: (same comment on the RELEASE_3_16 PR): could we perhaps assign the character string in line 131 to a variable `sfx` (or something for file name suffix), and use `length(sfx)` here? I feel like the `2` isn't very explicit or robust. E.g., I added another suffix not changing the 2 and it runs silently, but samples/files get mixed up... just to keep things clear and safe for the future.

File: tests/testthat/test_read10xVisium.R [hunk 2]
```diff
@@ -72,3 +72,33 @@ test_that("adding 'outs/' directory works correctly for one or more samples", {
...
+    
+    # correct number of spatial locations
+    expect_equal(
+        nrow(spatial_coords_multi), 
+        2 * nrow(spatial_coords_1) + 2 * nrow(spatial_coords_2))
+    
+    # spatial coordinates are in correct order
+    expect_identical(
+        spatial_coords_multi, 
+        rbind(spatial_coords_1, spatial_coords_2, spatial_coords_1, spatial_coords_2))
```
Comment (HelenaLC) [inline tests/testthat/test_read10xVisium.R:103]: line break after `spatial_coords_2,` please to keep the 80-char limits 🙏

File: R/read10xVisium.R (hunk 1 above)
Comment (lmweber) [reply R/read10xVisium.R:128]: Ah yes this could work, thanks. I'll try it.

**Comment by lmweber** [comment]
Thanks for the suggestions @HelenaLC . I agree using the suffix variable/vector is more robust. I have added both updates (also the 80 chars line length) in both PRs now. Hopefully all the checks above will pass again too. Thank you!

**Comment by lmweber** [comment]
I'll approve and merge / push this PR too since @HelenaLC has approved the same updates in the release branch `RELEASE_3_16`. Thank you!

**Review by lmweber — APPROVED** [review]

**Pull request merged**